### ISSUE TITLE
Conflict with ember-simple-auth: On 4.12.0, ember-data appears to lead to ember-simple-auth sessions to be invalidated right after login

### ISSUE THREAD

**Issue by spuxx1701** [emberjs/data#8609]
Hey there! This is somewhat of a draft since I hadn't had the time yet to dig into what's going on exactly. The original issue can be found here on the `ember-simple-auth` repository: https://github.com/mainmatter/ember-simple-auth/issues/2533

### Behavior

After logging in, the session of `ember-simple-auth` is being invalidated right away. Obviously, this was pinned on the latest major release of `ember-simple-auth`, but I believe I managed to pin it down to `4.12.0` of `ember-data` today (the version of `ember-simple-auth` doesn't seem to matter).

### Versions

Versions to reproduce:
```json
"ember-data": "~4.12.0",
"ember-simple-auth": "^4.2.2",
```
Or:
```json
"ember-data": "~4.12.0",
"ember-simple-auth": "^5.0.0",
```

### Minimal reproduction

I'm currently on vacation so I doesn't have the time to provide a minimal reproduction, but I don't mind trying to get into it once I'm back and have some more time. This issue is mainly to let you guys know as well and serve as a reminder to me. :smile: Providing a minimal reproduction might be tricky though since I think some sort of authentication service is required. I think I'll start with the diff of `4.12.0` and debug.

**Comment by spuxx1701** [comment]
Maybe someone knows right away what's happening, so feel free to participate. :star2:

**Comment by Jopie01** [comment]
I'm also in the process to upgrade an app to the newest versions and also hit this. But I think this has nothing to do with `ember-data` because I go the error
```
Uncaught Error: Could not find module `@ember/polyfills` imported from `ember-simple-auth/utils/assign`
```
Upgrading to `ember-simple-auth 6.0.0-rc.1` fixed the issue for me, but maybe looking for the `@ember/polyfills` package will also do the trick.

**Comment by spuxx1701** [comment]
@Jopie01 that error sounds like it is unrelated to this issue tbh. I never ran into an error message. Did you observe the same behavior with the session being invalidated after login?

**Comment by Jopie01** [comment]
> Did you observe the same behavior with the session being invalidated after login?

Yes that's why I replied in the first place. Added some `debugger` in my code and all of a sudden that error came up in the console.

**Comment by Jopie01** [comment]
Well, you could also try the `6.0.0-rc.1` version of `ember-simple-auth` and see if it fixed the problem.

**Comment by spuxx1701** [comment]
> Well, you could also try the `6.0.0-rc.1` version of `ember-simple-auth` and see if it fixed the problem.

I can, but I don't think `ember-simple-auth` is the cause. At least in my case, I really was able to narrow it down to `ember-data` 4.12.0. I updated each dependency carefully and tested different versions thorouhly this afternoon.

**Comment by runspired** [comment]
@spuxx1701 it's possible simple-auth was depending on a transient dep that ember-data no longer brings, so I'd start with @Jopie01's suggestion.

**Comment by spuxx1701** [comment]
I did some testing and it at least in my case, the behavior was not caused by any of the polyfills, but instead by some custom changes (specifically querying a `session`) model that had a custom adapter. I am a bit confused, though - even after updating to `4.12`, shouldn't the adapter continue to be used by default (otherwise I would consider this a breaking change)?

Of course I could switch to a `RequestManager`, but I would not expect having to do that when updating to a minor. Is there a flag I need to provide to tell `ember-data` to use the adapter?

**Comment by runspired** [comment]
@spuxx1701 it should still use your custom adapter. How is your store service defined?

**Comment by spuxx1701** [comment]
Hey @runspired! My custom store extends the ember-data store: https://github.com/spuxx1701/potber-client/blob/main/app/services/custom-store.ts

**Comment by Jopie01** [comment]
When extending the store, see #8565 what to do. That issue also provides more information about the road forward.

**Comment by BobrImperator** [comment]
I'd like to chip in to this comment https://github.com/emberjs/data/issues/8609#issuecomment-1553662428 that `ember-simple-auth` doesn't rely on ember-data at all.

Authenticators rely only on "fetch" to make authentication calls.

**Comment by BobrImperator** [comment]
https://github.com/emberjs/data/issues/8609#issuecomment-1604252037

It also appears that it could be some problem with Ember-Data 4.12.0 specifically that is later fixed in `5.0`.

I've recently added a  `ember-lts-4.12`  test scenario for ESA and it's the only scenario failing, even `ember-canary` and `ember-beta` scenarios are passing see:
https://github.com/mainmatter/ember-simple-auth/actions/runs/5375433398/jobs/9752032441

![image](https://github.com/emberjs/data/assets/15948633/41fa830f-de5e-40c5-ae00-b32016d24728)
![image](https://github.com/emberjs/data/assets/15948633/518d10d6-34aa-4b2f-83e9-864bc707bbee)

Please notice that our `test-app` also has fastboot setup that used to be broken for all versions after 4.12 before this PR https://github.com/mainmatter/ember-simple-auth/pull/2575.

**Comment by BobrImperator** [comment]
Ember Data released a Patch `4.12.1`, it fixed `ember-simple-auth` CI for `ember-lts-4.12` scenario.
I recommend testing out the patch in your apps.

**Comment by runspired** [comment]
@BobrImperator awesome! going to close this for now then.
